### PR TITLE
Wikidata link

### LINKED ISSUE #449: Use Wikidata ids as stable ids
Feature Request: Add redirect thanks to Wikidata IDs.

For example : 

http://world.openfoodfacts.org/category/Q7802 would redirect to http://world.openfoodfacts.org/category/bread

http://world.openfoodfacts.org/country/Q38 would redirect to http://world.openfoodfacts.org/country/Italy

### PR DESCRIPTION
**Pull request by hangy**
**Description:**
Would allow links like http://world.openfoodfacts.org/category/wikidata:en:Q7802 to display http://world.openfoodfacts.org/category/breads, based on the present linked data.

**Related issues and discussion:** Fixes #449

### CODE REVIEW

**Comment by hangy** [comment]
Note: Theoretically, it should also work for [http://world.openfoodfacts.org/category/https%3A%2F%2Fwww.wikidata.org%2Fwiki%2FQ1197400](http://world.openfoodfacts.org/category/https%3A%2F%2Fwww.wikidata.org%2Fwiki%2FQ1197400), but for some reason, the 2nd slash gets lost somewhere (maybe in Apache or the CGI perl module).

File: lib/ProductOpener/Tags.pm [hunk 1]
```diff
@@ -2541,52 +2543,14 @@ sub canonicalize_taxonomy_tag($$$)
...
-				}
-
-				$matched_tagid = $canon_tagid;
-			}
-		}
-
-		if (defined $matched_tagid) {
-			return $matched_tagid;
-		}
+	my $linked_data_tag = canonicalize_taxonomy_tag_linkeddata($tagtype, $tag);
```
Comment (stephanegigandet) [inline lib/ProductOpener/Tags.pm]: In what circumstances would canonicalize_taxonomy_tag() be called with a linked data or weblink identifier?

When they are used in URLs like in http://world.openfoodfacts.org/category/wikidata:en:Q7802 then the canonicalization should already have been done through analyze_request()

canonicalize_taxonomy_tag() is used quite extensively internally, with input values that I think are never wikidata ids, so I would much prefer that we remove any extra subroutine calls from it.

Comment (hangy) [reply lib/ProductOpener/Tags.pm]: I'm not sure about the issue number, but the link detection was built in to `canonicalize_taxonomy_tag` to support users entering Wikidata ids or URLs on the website or the API when adding ie. category tags.

Comment (stephanegigandet) [reply lib/ProductOpener/Tags.pm]: I think we should remove that support. I have not seen anyone (or anything) using this. And meanwhile, it adds processing / memory usage for tons of calls to that function, on all taxonomies. E.g. for ingredient parsing, we call canonicalize_taxonomy_tag for each ingredient in the ingredient list. We could keep the support for the URLs (although I'm not really sure who would use it and in what circumstances), but remove for input values to tags.

**Comment by VaiTon** [comment]
@stephanegigandet what shall we do?

**Pull request merged**